src/main/java/com/google/api/codegen/transformer/PathTemplateTransformer.java

```diff
@@ -43,6 +63,150 @@
     return pathTemplates;
   }
 
+  public List<ResourceNameView> generateResourceNames(SurfaceTransformerContext context) {
+    return generateResourceNames(context, context.getApiConfig().getResourceNameConfigs().values());
+  }
+
+  public List<ResourceNameView> generateResourceNames(
+      SurfaceTransformerContext context, Iterable<ResourceNameConfig> configs) {
+    List<ResourceNameView> resourceNames = new ArrayList<>();
+    int index = 1;
+    for (ResourceNameConfig config : configs) {
+      switch (config.getResourceNameType()) {
+        case SINGLE:
+          resourceNames.add(
+              generateResourceNameSingle(context, index, (SingleResourceNameConfig) config));
+          break;
+        case ONEOF:
+          resourceNames.add(
+              generateResourceNameOneof(context, index, (ResourceNameOneofConfig) config));
+          break;
+        case FIXED:
+          resourceNames.add(
+              generateResourceNameFixed(context, index, (FixedResourceNameConfig) config));
+          break;
+        default:
+          throw new IllegalStateException("Unexpected resource-name type.");
+      }
+      index += 1;
+    }
+    return resourceNames;
+  }
+
+  private ResourceNameSingleView generateResourceNameSingle(
+      SurfaceTransformerContext context, int index, SingleResourceNameConfig config) {
+    SurfaceNamer namer = context.getNamer();
+    ResourceNameSingleView.Builder builder = ResourceNameSingleView.newBuilder();
+    builder.typeName(namer.getResourceTypeName(config));
```

Comment:
Indeed there is... I have been using the non-fluent api in general, since many builders have quite a few parameters. I personally haven't been forcing anyone either way.